### DISCUSSION TITLE
I need help getting results from Scalene

### DISCUSSION THREAD

**Discussion by vtnate** [plasma-umass/scalene#722] [Q&A]
I have a package that I want to use Scalene to help me understand memory & cpu usage: https://github.com/NREL/ThermalNetwork

I've tried a number of ways to get Scalene to work:
1. The primary entry point to this library is through its CLI. I can't get Scalene to work when calling the CLI. Is that expected?
2. I've installed the VSCode extension and ran it as directed, which doesn't work because I need user input via the CLI
3. I've run pytest via scalene and got the expected pytest output (see below), but Scalene says it didn't have enough time to profile. The test takes more than 1 second to run, so I don't understand what's happening.

Can you help me use this tool? I'd love to understand it better!

Thank you

 ```
❯ python3 -m scalene --- -m pytest thermalnetwork/tests/test_network.py                                                                                                                                                                  [09:51:16]
================================================================================================================ test session starts ================================================================================================================
platform darwin -- Python 3.11.4, pytest-7.4.3, pluggy-1.3.0
rootdir: /Users/nmoore/github/ThermalNetwork
configfile: pyproject.toml
plugins: anyio-4.0.0, cov-4.1.0, typeguard-4.1.5
collected 2 items

thermalnetwork/tests/test_network.py .s                                                                                                                                                                                                       [100%]

=========================================================================================================== 1 passed, 1 skipped in 14.03s ===========================================================================================================
Scalene: Program did not run for long enough to profile.
>>> elapsed time 16s
```

**Comment by AvdN** [comment]
I had a similar problem when calling a program that I wrote myself. When I passed the script that belongs to the program as parameter to scalene, it just evaluates the code in that script. As that does only call `__main__.main()` in the original code, it uses a very small amount of time in there from the total program run-time and I think that is what causes the "program did not run long enough".

What helped for me was running `scalene --profile-interval 10 --profile-all --- /path/to/script` to make sure things were working, and then removeing `--profile-interval 10`, to only get the total overview.

It would be nice if scalene suggests this based on the not finding enough data although a certain minimum amount of wall-time having passed.

**Reply by vtnate** [reply]
Thank you for your thoughts! Unfortunately that solution didn't work for me. I ran scalene with PyTest again, and got the same successful pytest output and no Scalene output of any sort.

I am frustrated at my inability to get this to work, as it seems like Scalene has great promise. Can someone from Scalene, maybe @emeryberger, help me use this properly?
- The [documentation for pytest](https://github.com/plasma-umass/scalene#asked-questions) seems outdated or otherwise not applicable.
- I also created a script that calls my code, and ran it with the profile-interval as you suggested. The command `scalene --profile-interval 10 --profile-all --- ./scalene_test.py` generated a huge amount of output to stdout, but no profile.json file.

**Comment by emeryberger** [comment]
Please comment on this issue!

https://github.com/plasma-umass/scalene/issues/760
